pkg/provisioner/helpers.go

```diff
@@ -236,6 +236,35 @@ func storageClassForObjectBucket(ob *v1alpha1.ObjectBucket, c kubernetes.Interfa
 	return class, nil
 }
 
+func addLabels(obj metav1.Object, newLabels map[string]string) {
+	labels := obj.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	for k, v1 := range newLabels {
+		if v2, ok := labels[k]; ok {
+			log.Info("key already exists in labels", k, v2, "overwritten by", v1)
+		}
+		labels[k] = v1
+	}
+	obj.SetLabels(labels)
+}
+
+func addFinalizers(obj metav1.Object, newFilalizers []string) {
+	finalizers := obj.GetFinalizers()
+	finalizerMap := make(map[string]struct{})
+	for _, f := range finalizers {
+		finalizerMap[f] = struct{}{}
+	}
+
+	for _, f := range newFilalizers {
```

Comment:
why does the lib override (with a warning) user labels that conflict with ours, but we don't override user finalizers that conflict?

Reply:
I'm not sure why the `finalizerMap` is needed? Can't we append the lib finalzers array to the existing finalizers array returned by `getFinalizers`?

Reply:
I think the lib should override conflicting finalizers also.

Reply:
If we simply add a finalizer, the value will be duplicated if retries happen.  For this reason, we append only when there is no finalizer in the list.

Reply:
Good point about dealing with retries and I agree that finalizers map supports this. Since labels are a map and finalizers are an array, the logic will not be symmetric between the two funcs. However, won't retries spam the log for adding labels when the key already exists from try N-1?

Reply:
Indeed. How about logging only if label value is changed.